### PR TITLE
DecryptAndDecompressTar: return ReadCloser instead of taking Writer

### PR DESCRIPTION
**Pull request by serprex**
This allows tryExtractFiles to drop io.Pipe in favor of feeding the decompression reader directly into the tar interpreter

In production we've seen a few databases which'll continuously fail to download some parts from their base backups. This issue persists across base backups & forks

This issue has been reproduced across:
- Before & after decompression refactor
- With lz4, lz4/v4, & brotli
- Before & after azure sdk update

The issue is currently mitigated using curl & tar to complete missing parts

After this change the issue was resolved

### CODE REVIEW

File: internal/extract.go [hunk 1]
```diff
@@ -218,34 +171,30 @@ func tryExtractFiles(files []ReaderMaker,
 			tracelog.ErrorLogger.Println(err)
 			return files //Should never happen, but if we are asked to cancel - consider all files unfinished
 		}
-		err = writingSemaphore.Acquire(downloadingContext, 1)
```
Comment (usernamedt) [inline internal/extract.go:221]: Why did you remove the `writingSemaphore` logic?

**Review by usernamedt — CHANGES_REQUESTED** [review]

File: internal/extract.go (hunk 1 above)
Comment (serprex) [reply internal/extract.go:221]: There is now only 1 goroutine which decompresses & writes, instead of sending decompresssed bytes to pipe which gets written

Comment (usernamedt) [reply internal/extract.go:221]: > There is now only 1 goroutine which decompresses & writes, instead of sending decompresssed bytes to pipe which gets written
> 
> These are all still capped by a `semaphore.NewWeighted(int64(downloadingConcurrency))`

Oh, looks like I've missed it...

**Review by usernamedt — APPROVED** [review]
